Sources/Clients/OnLedgerEntitiesClient/Helpers/OnLedgerEntitiesClient+CreateEntity.swift

```diff
@@ -0,0 +1,679 @@
+			let stakeClaimNFT: OnLedgerEntity.OwnedNonFungibleResource? = {
+				guard let stakeClaimNFT = stake.stakeClaimResource, stakeClaimNFT.nonFungibleIdsCount > 0 else {
+					return nil
+				}
+				return stakeClaimNFT
+			}()
+
+			if stakeUnitResource != nil || stakeClaimNFT != nil {
+				return OnLedgerEntity.Account.RadixNetworkStake(
+					validatorAddress: stake.validatorAddress,
+					stakeUnitResource: stakeUnitResource,
+					stakeClaimResource: stakeClaimNFT
+				)
+			}
+			return nil
+		}
+
+		return .init(radixNetworkStakes: stakes, poolUnits: poolUnits)
+	}
+}
+
+extension OnLedgerEntitiesClient {
+	public struct OwnedStakeDetails: Hashable, Sendable {
+		public let validator: OnLedgerEntity.Validator
+		public let stakeUnitResource: ResourceWithVaultAmount?
+		public let stakeClaimTokens: NonFunbileResourceWithTokens?
+		public let currentEpoch: Epoch
+	}
+
+	public struct OwnedResourcePoolDetails: Hashable, Sendable {
+		public let address: ResourcePoolAddress
+		public let poolUnitResource: ResourceWithVaultAmount
+		public let xrdResource: ResourceWithVaultAmount?
+		public let nonXrdResources: [ResourceWithVaultAmount]
+	}
+
+	public struct ResourceWithVaultAmount: Hashable, Sendable {
+		public let resource: OnLedgerEntity.Resource
+		public let amount: RETDecimal
+	}
+
+	public struct NonFunbileResourceWithTokens: Hashable, Sendable {
+		public let resource: OnLedgerEntity.Resource
+		public let tokens: [OnLedgerEntity.NonFungibleToken]
+	}
+}
+
+extension Array where Element == OnLedgerEntity.OwnedFungibleResource {
+	func sorted() async -> OnLedgerEntity.OwnedFungibleResources {
+		@Dependency(\.gatewaysClient) var gatewaysClient
+
+		var xrdResource: OnLedgerEntity.OwnedFungibleResource?
+		var nonXrdResources: [OnLedgerEntity.OwnedFungibleResource] = []
+
+		let networkId = await gatewaysClient.getCurrentNetworkID()
+
+		for resource in self {
+			if resource.resourceAddress.isXRD(on: networkId) {
+				xrdResource = resource
+			} else {
+				nonXrdResources.append(resource)
+			}
+		}
+
+		let sortedNonXrdResources = nonXrdResources.sorted { lhs, rhs in
+			if lhs.amount > .zero, rhs.amount > .zero {
+				return lhs.amount > rhs.amount // Sort descending by amount
+			}
+			if lhs.amount != .zero || rhs.amount != .zero {
+				return lhs.amount != .zero
+			}
+
+			if let lhsSymbol = lhs.metadata.symbol, let rhsSymbol = rhs.metadata.symbol {
+				return lhsSymbol < rhsSymbol // Sort alphabetically by symbol
+			}
+			if lhs.metadata.symbol != nil || rhs.metadata.symbol != nil {
+				return lhs.metadata.symbol != nil
+			}
+
+			if let lhsName = lhs.metadata.name, let rhsName = rhs.metadata.name {
+				return lhsName < rhsName // Sort alphabetically by name
+			}
+
+			return lhs.resourceAddress.address < rhs.resourceAddress.address // Sort by address
+		}
+
+		return .init(xrdResource: xrdResource, nonXrdResources: sortedNonXrdResources)
+	}
+}
+
+extension Array where Element == OnLedgerEntity.OwnedNonFungibleResource {
```

Comment:
Better to let `OnLedgerEntity.OwnedNonFungibleResource` conform to `Comparable` and thus this `func sorted()` can be removed, and reused on all types conforming to `Sequence` ?